### PR TITLE
Data kit error status code 400 #76

### LINKED ISSUE #76: Data kit error status code 400
We take as input sometimes `queryByMetric` where `from` >= `to`. Grafana can't query ranges where `to` less than `from`. Datakit should handle this cases instead of sending request and get 400 error.

To resolve it we should check `from` and `to` values:
 - throw error if `from` > `to`
 - log warning if `from` = `to`

### PR DESCRIPTION
**Pull request by amper43**
fixes #76 

### Changes
- [x] throw error if from > to
- [x] log warning if from = to

### CODE REVIEW

File: src/grafana_service.ts [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
 import * as _ from 'lodash';
 
-
-export class GrafanaUnavailable extends Error {};
-export class DatasourceUnavailable extends Error {
-  public url: string;
+export class DataKitError extends Error {
+  constructor(
+    message: string,
+    public datasourceType?: string,
```
Comment (jonyrock) [inline src/grafana_service.ts]: do we have an `enum` for `datasourceType` ?

Comment (jonyrock) [reply src/grafana_service.ts]: oh, I see. Let's make it ones: https://github.com/CorpGlory/grafana-datasource-kit/blob/master/src/metrics/metric.ts#L3

File: src/grafana_service.ts [hunk 2]
```diff
@@ -1,16 +1,24 @@
...
+    public datasourceUrl?: string
+  ) {
+    super(message);
+  }
 };
 
+export class DataKitBadRange extends DataKitError {};
```
Comment (jonyrock) [inline src/grafana_service.ts]: lets rename it to `BadRange`

File: src/grafana_service.ts (hunk 1 above)
Comment (amper43) [reply src/grafana_service.ts]: added issue #78

**Pull request merged**